rgw/v2/lib/resource_op.py

```diff
@@ -125,6 +125,44 @@ def create_users(no_of_users_to_create, user_names=None, cluster_name="ceph"):
     return all_users_details
 
 
+def create_non_tenant_sub_users(no_of_users_to_create, no_of_swift_user_to_create):
+    """
+    This function is to create n users and m subusers on the cluster
+
+    Parameters:
+        no_of_users_to_create(int): users to create
+        no_of_swift_user_to_create(int): sub users to create
+
+    Returns:
+        all_users_details
+        sub_user_details
+    """
+    all_users_details = []
+    sub_user_details = []
+    for user_count in range(1, no_of_users_to_create + 1):
+        user_info = create_users(user_count)[-1]
+        log.info(user_info)
+        all_users_details.append(user_info)
+        log.info(f"creating swift user for {user_info['user_id']}")
+        for subuser_count in range(1, no_of_swift_user_to_create + 1):
+            sub_user_name = "swift_" + str(subuser_count)
+            sub_user = json.loads(
+                utils.exec_shell_cmd(
+                    f"radosgw-admin subuser create --uid {user_info['user_id']} --subuser={user_info['user_id']}:{sub_user_name} --access=full"
+                )
+            )
+            if not sub_user:
+                raise AssertionError(
+                    f"Failed to create subuser {sub_user_name} in user {user_info['user_id']}"
+                )
+            sub_user_info = {
+                "user_id": sub_user["swift_keys"][0]["user"],
+                "key": sub_user["swift_keys"][0]["secret_key"],
+            }
+            sub_user_details.append(sub_user_info)
+    return all_users_details, sub_user_details
```

Comment:
Just a thought,
if we create multiple user and subuser here in this method will there be any mapping between user and subuser?

shall we keep this method only for creation of subuser, lets use user name as parameter here?

Please correct me if I ma wrong